### PR TITLE
javadoc for random

### PR DESCRIPTION
**Pull request by JkSelf**
A subtask of https://github.com/intel-hadoop/chimera/issues/91
Complete the javadoc for Random

### CODE REVIEW

File: src/main/java/com/intel/chimera/random/JavaSecureRandom.java [hunk 1]
```diff
@@ -34,6 +34,10 @@
 
   private java.security.SecureRandom instance;
 
+  /**
+   * Constructs a {@link com.intel.chimera.random.JavaSecureRandom}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/JavaSecureRandom.java]: Better to left a full stop ·.· at the end.

File: src/main/java/com/intel/chimera/random/SecureRandomFactory.java [hunk 2]
```diff
@@ -29,12 +29,19 @@
...
 /**
- * The Factory for SecureRandom.
+ * This is the factory class used for SecureRandom class
  */
 public class SecureRandomFactory {
   public final static Logger LOG = LoggerFactory
       .getLogger(SecureRandomFactory.class);
 
+  /**
+   * Get a SecureRandom instance for specified props.
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/SecureRandomFactory.java]: Gets

File: src/main/java/com/intel/chimera/random/SecureRandomFactory.java [hunk 3]
```diff
@@ -29,12 +29,19 @@
     .CHIMERA_CRYPTO_SECURE_RANDOM_CLASSES_KEY;
 
 /**
- * The Factory for SecureRandom.
+ * This is the factory class used for SecureRandom class
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/SecureRandomFactory.java]: `SecureRandom class` can be `{@link SecureRandom}`

File: src/main/java/com/intel/chimera/random/JavaSecureRandom.java [hunk 4]
```diff
@@ -45,11 +49,20 @@ public JavaSecureRandom(Properties properties) {
     }
   }
 
+  /**
+   * Overrides {@link java.lang.AutoCloseable#close()}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/JavaSecureRandom.java]: Overrides {@link java.lang.AutoCloseable#close()}. For {@link JavaSecureRandom}, we don't need to recycle resource.

File: src/main/java/com/intel/chimera/random/JavaSecureRandom.java [hunk 5]
```diff
@@ -45,11 +49,20 @@ public JavaSecureRandom(Properties properties) {
...
+   *
+   */
   @Override
   public void close() {
     // do nothing
   }
 
+  /**
+   * Overrides {@link com.intel.chimera.random.SecureRandom#nextBytes(byte[])}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/JavaSecureRandom.java]: Overrides {@link com.intel.chimera.random.SecureRandom#nextBytes(byte[])}. Generates random bytes and places them into a user-supplied byte array.  The number of random bytes produced is equal to the length of the byte array.

File: src/main/java/com/intel/chimera/random/OpensslSecureRandom.java [hunk 6]
```diff
@@ -59,10 +59,17 @@
     }
   }
 
+  /**
+   * judge whether to need nativecode Load
+   */
   public static boolean isNativeCodeLoaded() {
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandom.java]: Judges whether loading native library successfully.

`@return true if loading native library successfully`

File: src/main/java/com/intel/chimera/random/OpensslSecureRandom.java [hunk 7]
```diff
@@ -82,14 +89,18 @@ public void nextBytes(byte[] bytes) {
     }
   }
 
+  /**
+   * Overrides {@link java.util.Random# setSeed()}
+   *
+   * @param seed the initial seed
+   */
   @Override
   public void setSeed(long seed) {
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandom.java]: Overrides {@link java.util.Random#setSeed()}. For {@link OpensslSecureRandom}, we don't need to set seed.

File: src/main/java/com/intel/chimera/random/OpensslSecureRandom.java [hunk 8]
```diff
@@ -82,14 +89,18 @@ public void nextBytes(byte[] bytes) {
...
     // Self-seeding.
   }
 
   /**
-   * Generates an integer containing the user-specified number of
-   * random bits (right justified, with leading zeros).
+   * Overrides {@link java.util.Random# next()}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandom.java]: Overrides {@link java.util.Random# next()}. Generates an integer containing the user-specified number of random bits (right justified, with leading zeros).

File: src/main/java/com/intel/chimera/random/OpensslSecureRandom.java [hunk 9]
```diff
@@ -59,10 +59,17 @@
...
+  /**
+   * judge whether to need nativecode Load
+   */
   public static boolean isNativeCodeLoaded() {
     return nativeEnabled;
   }
 
+  /**
+   * Constructs a {@link com.intel.chimera.random.OpensslSecureRandom}
+   * @param props the configuration properties
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandom.java]: Better to leave a blank line here.

File: src/main/java/com/intel/chimera/random/JavaSecureRandom.java [hunk 10]
```diff
@@ -34,6 +34,10 @@
 
   private java.security.SecureRandom instance;
 
+  /**
+   * Constructs a {@link com.intel.chimera.random.JavaSecureRandom}
+   * @param properties  the configuration properties
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/JavaSecureRandom.java]: One blank space between `properties` and `the`

File: src/main/java/com/intel/chimera/random/JavaSecureRandom.java [hunk 11]
```diff
@@ -45,11 +49,20 @@ public JavaSecureRandom(Properties properties) {
     }
   }
 
+  /**
+   * Overrides {@link java.lang.AutoCloseable#close()}
+   *
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/JavaSecureRandom.java]: No black line if there is no @param or @return

File: src/main/java/com/intel/chimera/random/OpensslSecureRandom.java [hunk 12]
```diff
@@ -82,14 +89,18 @@ public void nextBytes(byte[] bytes) {
     }
   }
 
+  /**
+   * Overrides {@link java.util.Random# setSeed()}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandom.java]: Overrides {@link java.util.Random#setSeed()}. For {@link OpensslSecureRandom}, we don't need to set seed.

File: src/main/java/com/intel/chimera/random/OpensslSecureRandom.java [hunk 13]
```diff
@@ -112,6 +123,10 @@ final protected int next(int numBits) {
     return next >>> (numBytes * 8 - numBits);
   }
 
+  /**
+   * Overrides {@link java.lang.AutoCloseable#close()}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandom.java]: Overrides {@link java.lang.AutoCloseable#close()}. Closes the OS stream if native enabled.

File: src/main/java/com/intel/chimera/random/OpensslSecureRandomNative.java [hunk 14]
```diff
@@ -17,8 +17,20 @@
  */
 package com.intel.chimera.random;
 
+/**
+ * JNI interface of {@link SecureRandom} implementation. The native method in this class is defined in OpensslSecureRandomNative.h(genereted by javah).
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandomNative.java]: The line is too long, better to use two or three lines.

File: src/main/java/com/intel/chimera/random/OpensslSecureRandomNative.java [hunk 15]
```diff
@@ -17,8 +17,20 @@
...
+   * Declares a native method to initialize SR.
+   */
   public native static void initSR();
 
+  /**
+   * Generates a user-specified number of random bytes.
+   * It's thread-safe.
+   *
+   * @param bytes the array to be filled in with random bytes.
+   */
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OpensslSecureRandomNative.java]: Add @return

File: src/main/java/com/intel/chimera/random/OsSecureRandom.java [hunk 16]
```diff
@@ -100,6 +116,11 @@ synchronized protected int next(int nbits) {
     return n & (0xffffffff >> (32 - nbits));
   }
 
+
+  /**
+   * Overrides {@link java.lang.AutoCloseable# close()}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OsSecureRandom.java]: Overrides {@link java.lang.AutoCloseable#`no blank here`close()}. Closes the OS stream.

File: src/main/java/com/intel/chimera/random/OsSecureRandom.java [hunk 17]
```diff
@@ -58,6 +58,10 @@ private void fillReservoir(int min) {
     }
   }
 
+  /**
+   * Constructs a {@link com.intel.chimera.random.OsSecureRandom}
+   * @param props  the configuration properties
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OsSecureRandom.java]: One blank between props and the

File: src/main/java/com/intel/chimera/random/OsSecureRandom.java [hunk 18]
```diff
@@ -77,6 +81,11 @@ public OsSecureRandom(Properties props) {
     }
   }
 
+  /**
+   * Overrides {@link com.intel.chimera.random.SecureRandom#nextBytes(byte[])}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OsSecureRandom.java]: Overrides {@link com.intel.chimera.random.SecureRandom#nextBytes(byte[])} `Add some comments for the method from inteface`

File: src/main/java/com/intel/chimera/random/OsSecureRandom.java [hunk 19]
```diff
@@ -90,6 +99,13 @@ synchronized public void nextBytes(byte[] bytes) {
     }
   }
 
+  /**
+   * Overrides {@link java.util.Random# next()}
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/random/OsSecureRandom.java]: Overrides {@link java.util.Random# next()}. `Add some comments for the method from inteface`

**Comment by sundapeng** [comment]
Thank @JkSelf for your contribution. I left some comments.

File: src/main/java/com/intel/chimera/random/JavaSecureRandom.java (hunk 1 above)
Comment (JkSelf) [inline src/main/java/com/intel/chimera/random/JavaSecureRandom.java]: Yes, I have added the full stop . at the ned, thank you for your comment

File: src/main/java/com/intel/chimera/random/SecureRandomFactory.java (hunk 2 above)
Comment (JkSelf) [inline src/main/java/com/intel/chimera/random/SecureRandomFactory.java]: I have changed the Get to Gets, thank you for your comment

**Comment by JkSelf** [comment]
Thanks for  @sundapeng comments. I have fixed.

File: src/main/java/com/intel/chimera/input/StreamInput.java [hunk 20]
```diff
@@ -35,6 +35,15 @@ public StreamInput(InputStream inputStream, int bufferSize) {
     this.bufferSize = bufferSize;
   }
 
+  /**
```
Comment (sundapeng) [inline src/main/java/com/intel/chimera/input/StreamInput.java]: Better to keep this pr is only related to `random`. How about move `StreamInput` to next pr?

**Comment by sundapeng** [comment]
Thank @JkSelf . Only one minor comment left, others are LGTM.

File: src/main/java/com/intel/chimera/input/StreamInput.java (hunk 20 above)
Comment (JkSelf) [inline src/main/java/com/intel/chimera/input/StreamInput.java]: Thanks for @sundapeng comment. I have fixed.

**Comment by sundapeng** [comment]
+1, Thank @JkSelf

**Pull request merged**